agent_api_rest/src/issuance/credential_issuer/token_status_list.rs

```diff
@@ -0,0 +1,211 @@
+                None
+            }
+        })
+        .collect();
+
+    // This block ensures that the remaining empty 30% of a status list is filled with random values.
+    // This block works in tandem with the part of `fn patch_credential` which only fills 70% of a status list.
+    used_indices = task::spawn_blocking(move || -> Result<Vec<CredentialStatus>, PublicError> {
+        let mut indices = used_indices.clone();
+
+        let mut rng = rand::rng();
+        while indices.len() < amount_indices {
+            let random_index = rng.random_range(lower_bound..upper_bound);
+            if !indices
+                .iter()
+                .any(|credential_status| credential_status.index == random_index)
+            {
+                let status_type = rng.random_range(0..2);
+                indices.push(CredentialStatus {
+                    index: random_index,
+                    status: status_type.try_into().map_err(|_| PublicError::InternalServerError)?,
+                });
+            }
+        }
+
+        Ok(indices)
+    })
+    .await
+    .map_err(|_| PublicError::InternalServerError)??;
+
+    used_indices.sort_by_key(|credential_status| credential_status.index);
+
+    let mut status_list = StatusList {
+        status_size: Bits::try_from(STATUSTYPESIZE).map_err(|_| PublicError::InternalServerError)?,
+        ..Default::default()
+    };
+    status_list
+        .pack_statuses_into_bytes(used_indices.iter().map(|s| s.status).collect::<Vec<StatusType>>())
+        .map_err(|_| PublicError::InternalServerError)?;
+
+    let mut sub_url = config().ietf_oauth_token_status_list_uri.clone();
+    sub_url
+        .path_segments_mut()
+        .map_err(|_| PublicError::InternalServerError)?
+        .push(&status_list_number.to_string());
+    let status_list_claims = StatusListTokenClaims {
+        sub: sub_url.to_string(),
+        iat: chrono::Utc::now().timestamp(),
+        exp: None,
+        ttl: None,
+        encoded_status_list: EncodedStatusList::try_from(status_list).map_err(|_| PublicError::InternalServerError)?,
+    };
+
+    let mut status_list_token = StatusListToken {
+        claims: status_list_claims,
+        ..Default::default()
+    };
+    status_list_token.header.alg = get_preferred_signing_algorithm();
+    let default_did_method = get_preferred_did_method().to_string();
+
+    let jwt_token = encode(
+        state.signer,
+        status_list_token.header,
+        status_list_token.claims,
+        &default_did_method,
+    )
+    .await
+    .map_err(|_| PublicError::InternalServerError)?;
+
+    let compressed_jwt_token = compress_gzip(&jwt_token).map_err(|_| PublicError::InternalServerError)?;
+
+    Ok((
+        StatusCode::OK,
+        [
+            (
+                header::CONTENT_TYPE,
+                StatusListTokenResponseType::Jwt.to_string().as_str(),
+            ),
+            (header::CONTENT_ENCODING, "gzip"),
+        ],
+        compressed_jwt_token,
+    )
```

Comment:
I'm not really familiar with this `gzip` content encoding but currently it causes this line to panic in our middleware: `info!("Response Body: {}", std::str::from_utf8(chunk).unwrap());`

So to fix that you need to get rid of that unwrap (which was already very bad in the first place) by only logging the message if `std::str::from_utf8(chunk)` returns `Ok`